packages/loopring_v3/contracts/iface/ExchangeData.sol

```diff
@@ -168,27 +114,20 @@ library ExchangeData
         return 21888242871839275222246405745257275088548364400416034343698204186575808495617;
     }
 
-    function MAX_PROOF_GENERATION_TIME_IN_SECONDS() internal pure returns (uint32) { return 14 days; }
-    function MAX_GAP_BETWEEN_FINALIZED_AND_VERIFIED_BLOCKS() internal pure returns (uint32) { return 1000; }
     function MAX_OPEN_DEPOSIT_REQUESTS() internal pure returns (uint16) { return 1024; }
     function MAX_OPEN_WITHDRAWAL_REQUESTS() internal pure returns (uint16) { return 1024; }
-    function MAX_AGE_UNFINALIZED_BLOCK_UNTIL_WITHDRAW_MODE() internal pure returns (uint32) { return 21 days; }
     function MAX_AGE_REQUEST_UNTIL_FORCED() internal pure returns (uint32) { return 14 days; }
     function MAX_AGE_REQUEST_UNTIL_WITHDRAW_MODE() internal pure returns (uint32) { return 15 days; }
     function MAX_TIME_IN_SHUTDOWN_BASE() internal pure returns (uint32) { return 30 days; }
     function MAX_TIME_IN_SHUTDOWN_DELTA() internal pure returns (uint32) { return 1 seconds; }
     function TIMESTAMP_HALF_WINDOW_SIZE_IN_SECONDS() internal pure returns (uint32) { return 7 days; }
     function MAX_NUM_TOKENS() internal pure returns (uint) { return 2 ** 8; }
-    function MAX_NUM_ACCOUNTS() internal pure returns (uint) { return 2 ** 20 - 1; }
-    function MAX_TIME_TO_DISTRIBUTE_WITHDRAWALS() internal pure returns (uint32) { return 14 days; }
-    function MAX_TIME_TO_DISTRIBUTE_WITHDRAWALS_SHUTDOWN_MODE() internal pure returns (uint32) {
-        return MAX_TIME_TO_DISTRIBUTE_WITHDRAWALS() * 10;
-    }
+    function MAX_NUM_ACCOUNTS() internal pure returns (uint) { return 2 ** 24 - 1; }
     function FEE_BLOCK_FINE_START_TIME() internal pure returns (uint32) { return 6 hours; }
     function FEE_BLOCK_FINE_MAX_DURATION() internal pure returns (uint32) { return 6 hours; }
-    function MIN_GAS_TO_DISTRIBUTE_WITHDRAWALS() internal pure returns (uint32) { return 150000; }
     function MIN_AGE_PROTOCOL_FEES_UNTIL_UPDATED() internal pure returns (uint32) { return 1 days; }
     function GAS_LIMIT_SEND_TOKENS() internal pure returns (uint32) { return 60000; }
+    function MAX_TOTAL_TOKEN_BALANCE() internal pure returns (uint) { return 2 ** 96 - 1; }
```

Comment:
Moved the check to the deposit contract for now because it needs to be checked against the total amount stored in the exchange (which is currently kept only in the deposit contract). Depending on how the deposit contract still evolves it may make sense to still store total amounts (as stored in the Merkle tree) in the exchange contract or not.

The check isn't necessary in `approveConditionalTransfer` because even if the value would be higher the total amount could be transferred in multiple transfers. But it doesn't make much sense that it would be higher because it would the more than can be stored totally in the exchange with the current checks.